12.0 odoo_backup_sh bugfixes and doc fixes

em230418 opened the pull request.

yelizariev commented on odoo_backup_sh/controllers/main.py:
```diff
@@ -131,6 +131,16 @@ def update_cloud_params(cls, cloud_params, redirect, call_from):
                 ).unlink()
         return cloud_params
 
+    @http.route('/web/database/backup', type='http', auth='none')
```
It cannot be auth=none here

yelizariev commented on odoo_backup_sh/controllers/main.py:
```diff
@@ -131,6 +131,16 @@ def update_cloud_params(cls, cloud_params, redirect, call_from):
+        }
+        cloud_params = self.get_cloud_params(request.httprequest.url, call_from='frontend')
+        backup_object = BackupCloudStorage.get_object(cloud_params, backup_file_name)
+        return request.make_response(backup_object['Body'].read(), headers)
```
Looks like you allow to anyone download a backup

yelizariev commented on odoo_backup_sh/doc/index.rst:
```diff
@@ -75,3 +75,14 @@ Usage
+
+* Proceed to the Backup list: ``/web/database/backups``
+* Click on ``Download`` button and wait until download completes
+* Backup is encrypted (it has `.enc` extension), extract it using `gpg` utility. As example:
```
``If`` is missed?

Mirgalimova replied:
fixed

yelizariev commented on odoo_backup_sh/doc/index.rst:
```diff
@@ -75,3 +75,14 @@ Usage
+Downloading backups
+===================
+
+* Proceed to the Backup list: ``/web/database/backups``
```
Downloading must be done in backend

Mirgalimova replied:
@em230418 can you fix it?

Mirgalimova replied:
@em230418 please check)

yelizariev commented on odoo_backup_sh_dropbox/doc/index.rst:
```diff
@@ -68,3 +68,15 @@ Usage
+
+* Proceed to the Backup list: ``/web/database/backups``
+* Click on ``Download`` button and wait until download completes
+* Backup is encrypted (it has `.enc` extension), extract it using `gpg` utility. As example:
```
In dropbox you can download directly from dropbox.

yelizariev commented on odoo_backup_sh/doc/index.rst:
````diff
@@ -75,3 +75,14 @@ Usage
+
+```
+gpg --output OUTPUT_FILENAME.zip --decrypt INPUT_FILENAME.zip.enc
+```
````
Add a note like "Type the password when it will be prompted"

Mirgalimova replied:
fixed

em230418 commented:
@Mirgalimova @yelizariev I removed downloading notes from odoo_backup_sh_dropbox and odoo_backup_sh_google_disk and changed note in odoo_backup_sh. It is planned, that instructions given in last one will work for 2 others.

em230418 commented on odoo_backup_sh/controllers/main.py:
```diff
@@ -131,6 +131,24 @@ def update_cloud_params(cls, cloud_params, redirect, call_from):
+                'Content-Disposition': 'attachment; filename="{0}"'.format(backup_filename),
+                'Content-Type': 'application/octet-stream',
+            }
+        )
```
> DId you try to open the link with demo user, for example?

Yes.
![FireShot Capture 13 - db - Odoo](https://user-images.githubusercontent.com/52414052/63001683-acb39200-be63-11e9-829c-275b5990d304.png)

Should I add admin check in this function and similar functions?

yelizariev replied:
We need to create new Access Groups and check them before providing the backup

yelizariev commented on odoo_backup_sh/security/security.xml:
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0"?>
+<odoo>
+    <record id="base.group_user" model="res.groups">
+        <field name="implied_ids" eval="[(4, ref('odoo_backup_sh.group_user'))]"/>
```
Wrong. You allow to see backups for all employee

yelizariev commented on odoo_backup_sh/models/odoo_backup_sh.py:
```diff
@@ -36,6 +36,32 @@
+def get_backup_by_id(env, backup_id):
+    backup = env['odoo_backup_sh.backup_info'].search([
+        ('id', '=', backup_id),
+    ])
```
It's the same as 
`` env['odoo_backup_sh.backup_info'].browse(backup_id)
``

yelizariev commented on odoo_backup_sh/views/odoo_backup_sh_views.xml:
```diff
@@ -91,6 +100,7 @@ License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
                         <field name="encrypted"/>
                         <field name="backup_size" widget="float"/>
                     </group>
+                    <button class="btn btn-dark" type="action" name="%(download_backup)d" groups="odoo_backup_sh.group_manager">Download backup</button>
```
Use more classic button please. Check example in built-in forms

yelizariev replied:
Also, the button has to be hidden for simulation records

em230418 commented:
> I'm not sure that this is correct. `len(backup)` probably returns len(ids), which is always 1 here, while the record may not exist.

https://github.com/odoo/odoo/blob/94a2c54fb50e39e3343f3183ead2121940ed0878/odoo/models.py#L4980-L4982

It does not always return 1. At least in 12.0
